gradio/routes.py

```diff
@@ -592,26 +588,29 @@ def robots_txt():
 ########
 
 
-def safe_join(directory: str, path: str) -> str | None:
-    """Safely path to a base directory to avoid escaping the base directory.
-    Borrowed from: werkzeug.security.safe_join"""
-    _os_alt_seps: List[str] = list(
-        sep for sep in [os.path.sep, os.path.altsep] if sep is not None and sep != "/"
-    )
+def is_relative_to(self, other):
```

Comment:
@abidlabs this is what you are looking for

Reply:
But `is_relative_to` relies on the pathlib's Path's `relative_to` method internally. So it will not work on Python 3.7, right?

Reply:
Yes, `relative_to ` was introduced a while ago

Reply:
Ah my apologies for the confusion